packages/create-remix/index.ts

```diff
@@ -285,56 +277,117 @@ async function templateStep(ctx: Context) {
     start: "Template copying...",
     end: "Template copied",
     while: async () => {
-      let destPath = path.resolve(process.cwd(), ctx.cwd);
-      await ensureDirectory(destPath);
-      await copyTemplate(template, destPath, {
+      await ensureDirectory(ctx.tempDir);
+      if (ctx.debug) {
+        debug(`Extracting to: ${ctx.tempDir}`);
+      }
+
+      let result = await copyTemplate(template, ctx.tempDir, {
         debug: ctx.debug,
         token: ctx.token,
         async onError(err) {
-          let cwd = process.cwd();
-          let removing = (async () => {
-            if (cwd !== destPath && !pathContains(cwd, destPath)) {
-              try {
-                await rm(destPath);
-              } catch (_) {
-                error("Oh no!", ["Failed to remove ", destPath]);
-              }
-            }
-          })();
-          if (ctx.debug) {
-            try {
-              await removing;
-            } catch (_) {}
-            throw err;
-          }
-
-          await Promise.all([
-            error(
-              "Oh no!",
-              err instanceof CopyTemplateError
-                ? err.message
-                : "Something went wrong. Run `create-remix --debug` to see more info.\n\n" +
-                    "Open an issue to report the problem at " +
-                    "https://github.com/remix-run/remix/issues/new"
-            ),
-            removing,
-          ]);
-
+          error(
+            "Oh no!",
+            err instanceof CopyTemplateError
+              ? err.message
+              : "Something went wrong. Run `create-remix --debug` to see more info.\n\n" +
+                  "Open an issue to report the problem at " +
+                  "https://github.com/remix-run/remix/issues/new"
+          );
           throw err;
         },
         async log(message) {
           if (ctx.debug) {
-            info(message);
+            debug(message);
             await sleep(500);
           }
         },
       });
 
-      await updatePackageJSON(ctx);
+      if (result?.localTemplateDirectory) {
+        ctx.tempDir = path.resolve(result.localTemplateDirectory);
+      }
```

Comment:
Added the optimization that if the template is a local directory on disk we just copy from there instead of "extracting" it to a temp dir.